src/main/java/com/vacation_bot/core/classification/ClassificationService.java

```diff
@@ -0,0 +1,74 @@
+package com.vacation_bot.core.classification;
+
+import com.vacation_bot.core.customization.CustomizedSentence;
+import com.vacation_bot.core.words.WordsService;
+import com.vacation_bot.shared.MessageHeaders;
+import com.vacation_bot.shared.SentenceClass;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Classify the input customized sentence.
+ */
+public class ClassificationService {
+
+    public WordsService wordsService;
+
+    public ClassificationService( WordsService aWordsService ) {
+        wordsService = aWordsService;
+    }
+
+    @ServiceActivator
+    public Message<CustomizedSentence> classifySentence( Message<CustomizedSentence> message ) {
+        SentenceClass classResult = processSentence( message.getPayload().getCustomizedSentence() );
+
+        return MessageBuilder.fromMessage( message )
+                .setHeader( MessageHeaders.SENTENCE_CLASS_HEADER, classResult.toString() )
+                .build();
+    }
+
+    private SentenceClass processSentence(List<String> sentenceWords ) {
+        Map<String, List<String>> wordsPerSentenceType = wordsService.calculateWordsWorth();
+
+        SentenceClass classResult = SentenceClass.UNKNOWN;
+        int matches = 0;
+
+        int classOneCount = countCommonElements( sentenceWords, wordsPerSentenceType.get( SentenceClass.REQUEST_DAYS_LEFT.toString() ) );
```

Comment:
We can reduce near identical blocks of code with something like this (pseudocode):
```
class2score = Map();
for (sent_type in SentenceClass) {
  class2score[sent_type] = countCommonElements(sentenceWords, .....);
}
classResult = GetKeyWithMaxValue(class2score);
```